core/class/CommandLineTool.go

```diff
@@ -185,6 +190,41 @@ func (tool *CommandLineTool) placeInputFilesToCommandExecDir() error {
 	return nil
 }
 
+// defineStdoutDestination
+func (tool *CommandLineTool) defineStdinSource() error {
+	if tool.Root.Stdin != "" {
+		stdin, err := tool.Command.StdinPipe()
+		if err != nil {
+			return err
+		}
+		defer stdin.Close()
+		stdinfilename := tool.Root.Stdin
+		if strings.HasPrefix(stdinfilename, "$(") && strings.HasSuffix(stdinfilename, ")") {
+			// Create JavaScript runtime
+			vm, err := tool.Root.Inputs.ToJavaScriptVM(tool.Command.Dir)
+			if err != nil {
+				return err
+			}
+			retval, err := vm.Run(stdinfilename[2 : len(stdinfilename)-1])
+			if err != nil {
+				return err
+			}
+			stdinfilename = retval.String()
+		}
+		//
+		stdinfilepath := filepath.Join(tool.Command.Dir, stdinfilename)
```

Comment:
IF it's possible to `Entry.Location` to be absolute file path, this `Join` could be a bug. Do we need `IsAbs` check?